### PR TITLE
Add 2020-05-13-immutable-keyword.md post to blog

### PR DESCRIPTION
**Pull request by franzihei**

### CODE REVIEW

File: _posts/2020-05-13-immutable-keyword.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+since immutables will not be stored in storage, but their values will be directly inserted into the runtime code.
+
+## How to use Immutables
+
+Immutable state variables can be declared using the ``immutable`` keyword. They cannot be read during contract creation, 
+but either have to be written to exactly once in the constructor or initialized directly in their declaration. 
+Runtime code can only read immutables, but not write to them.
+<br><br>
+Here is an example of ``immutable`` in action:
+```
```
Comment (chriseth) [inline _posts/2020-05-13-immutable-keyword.md:24]:
```suggestion
```solidity
```

Comment (chriseth) [reply _posts/2020-05-13-immutable-keyword.md:24]: (not sure if it works already)

File: _posts/2020-05-13-immutable-keyword.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+```
+
+## How Immutables Work
+
+While regular state variables are stored at a fixed offset in storage, this is not the case for immutables. Instead, 
+immutables are stored directly in the deployed bytecode, i.e. the runtime code will contain placeholders for every 
+read of an immutable and the creation code will insert the values of the immutables into these placeholders. 
+Effectively, this means that reading from an immutable at runtime will reduce to a plain ``PUSH32`` opcode.
+<br><br>
```
Comment (chriseth) [inline _posts/2020-05-13-immutable-keyword.md:59]:
```suggestion


```

File: _posts/2020-05-13-immutable-keyword.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+constant throughout the life-time of a deployed contract.
+The big advantage of immutables is that reading them is significantly cheaper than reading from regular state variables, 
+since immutables will not be stored in storage, but their values will be directly inserted into the runtime code.
+
+## How to use Immutables
+
+Immutable state variables can be declared using the ``immutable`` keyword. They cannot be read during contract creation, 
+but either have to be written to exactly once in the constructor or initialized directly in their declaration. 
+Runtime code can only read immutables, but not write to them.
+<br><br>
```
Comment (chriseth) [inline _posts/2020-05-13-immutable-keyword.md:22]:
```suggestion

```

**Comment by chriseth** [comment]
Do we aim for publishing this today so we can have another one tomorrow about the release?

File: _posts/2020-05-13-immutable-keyword.md (hunk 1 above)
Comment (franzihei) [reply _posts/2020-05-13-immutable-keyword.md:24]: I think it doesn't or was there an update on the blog config? See https://github.com/ethereum/solidity-blog/issues/34

**Comment by franzihei** [comment]
Yes, from my side we could publish this one today!

**Comment by chriseth** [comment]
I'll rebase this.

File: _posts/2020-05-13-immutable-keyword.md (hunk 1 above)
Comment (chriseth) [reply _posts/2020-05-13-immutable-keyword.md:23]: Actually let me try - if this does not break it, we do not have to add it later on.

**Comment by chriseth** [comment]
Did not break. I'll add some more line breaks into the example because the lines are too long.

**Pull request merged**